src/main/java/plugin/DataCleanerJobEntry.java

```diff
@@ -45,12 +51,28 @@
     @Override
     public Result execute(Result result, int nr) throws KettleException {
 
+        final LogChannelInterface log = Spoon.getInstance().getLog();
+
         final List<String> commands = new ArrayList<String>();
-        final String executableFilePath = environmentSubstitute(configuration.getExecutableFilename());
-        final File executableFile = new File(executableFilePath);
         final String outputFilename = environmentSubstitute(configuration.getOutputFilename());
-
-        commands.add(executableFilePath);
+        final DataCleanerSpoonConfiguration dataCleanerSpoonConfiguration = ModelerHelper
+                .getDataCleanerSpoonConfigurationOrShowError();
+        final String dcInstallationPath = dataCleanerSpoonConfiguration.getDataCleanerInstallationFolderPath()
+                + "/DataCleaner.jar";
+
+        // Assemble the class path for DataCleaner
+        commands.add(System.getProperty("java.home") + "/bin/java");
+        commands.add("-cp");
+        commands.add(environmentSubstitute(dcInstallationPath));
```

Comment:
I took it into consideration, but the code has little in common. The numbers of jar we give to the command line in ModelerHelper are 5 or so.